lib/mayaUsd/utils/utilSerialization.cpp

```diff
@@ -184,6 +184,20 @@ void setNewProxyPath(const MString& proxyNodeName, const MString& newValue)
         /*undo*/ false);
 }
 
+bool saveLayerWithFormat(
+    SdfLayerRefPtr     layer,
+    const std::string& potentialFilePath,
+    std::string        formatArg)
+{
+    const std::string& filePath
+        = potentialFilePath.empty() ? layer->GetRealPath() : potentialFilePath;
+
+    PXR_NS::SdfFileFormat::FileFormatArguments args;
+    args["format"] = formatArg.empty() ? usdFormatArgOption() : formatArg;
+
+    return layer->Export(filePath, "", args);
```

Comment:
Is there a way here we can check the current format of the filePath and if it hasn't changed we just call Save()?

Reply:
Done, but it was trickier than I anticipated.